lib/utils.pm

```diff
@@ -276,6 +277,16 @@ sub zypper_call {
     elsif ($dumb_term) {
         script_run("zypper -n $command | cat; echo $str-\${PIPESTATUS}-$redirect", 0);
     }
+    elsif ($retry) {
+        for (1 .. $retry) {
+            eval { script_run("zypper -n $command; echo $str-\$?-$redirect", 0); };
+            last unless ($@);
```

Comment:
I don't think the script_run fails - especially as it's called with ,0

the following wait_serial  fails. And none of your validation runs actually ran into this code. What you should do IMO is calling the function with $retry-- in case of failure. And have a validation run that actually fails - i.e. by injecting invalid urls

Reply:
Ye, it was wrong. I have this now, not sure if you will like it :)
I'm not sure if I get the $retry thing, but it is done as it is also for the diag messages

faulty url $ and faulty url with accepted faulty exitcode
http://10.100.12.155/tests/5629
http://10.100.12.155/tests/6061
passing tests
http://10.100.12.155/tests/5635
http://10.100.12.155/tests/5636
http://10.100.12.155/tests/5637
http://10.100.12.155/tests/5639